cloudinit/distros/__init__.py

```diff
@@ -908,6 +910,50 @@ def lock_passwd(self, name):
             util.logexc(LOG, "Failed to disable password for user %s", name)
             raise e
 
+    def unlock_passwd(self, name):
+        """
+        Unlock the password of a user, i.e., enable password logins
+        """
+        # passwd must use short '-u' due to SLES11 lacking long form '--unlock'
+        unlock_tools = (["passwd", "-u", name], ["usermod", "--unlock", name])
+        try:
+            cmd = next(tool for tool in unlock_tools if subp.which(tool[0]))
+        except StopIteration as e:
+            raise RuntimeError(
+                "Unable to unlock user account '%s'. No tools available. "
+                "  Tried: %s." % (name, [c[0] for c in unlock_tools])
+            ) from e
+        try:
+            _, err = subp.subp(cmd, rcs=[0, 3])
```

Comment:
Because we are falling back on a non-empty stderr due to exit 3 from passwd -u <user>, we are taking automated steps in cloud-init to essentially ignore the warning message about risky password policies that passwd is trying to show us with that exit 3
```
passwd: unlocking the password would result in a passwordless account.
You should set a password with usermod -p to unlock the password of this account.
```

Note that usermod -U <user> would also advise us against this too with a similar message:
```
usermod: unlocking the user's password would result in a passwordless account.
You should set a password with usermod -p to unlock this user's password
```

I really think we should make the admin be intentional in their cloud-config somehow if they really need as passwordless user.  As it is currently if a user is defined with cloud-config: `lock_passwd: false` and they mistakenly omitted a passwd/hashed_passwd/plain_text_passwd related key, cloud-init would happily setup that user with an empty password without any sort of notification about this potentially risky configuration. As mentioned in the review, if we can find a more clear expression of "I really want an empty password"  in cloud-config, such as providing a specific empty string (not None/null) for one of those password keys, I think that would make me more comfortable with the admin's intent.

Reply:
Addressed.